### PR TITLE
Fix #23919: Root message for new thread loaded from network

### PR DESCRIPTION
**Pull request by justjanne**
Type: Defect
Fixes: https://github.com/vector-im/element-web/issues/23919

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix #23919: Root message for new thread loaded from network ([\#2965](https://github.com/matrix-org/matrix-js-sdk/pull/2965)). Fixes vector-im/element-web#23919. Contributed by @justjanne.

### CODE REVIEW

**Review by gsouquet — APPROVED** [review]
Looks good to me!

File: src/models/thread.ts [hunk 1]
```diff
@@ -358,7 +358,15 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
...
-                await this.client.paginateEventTimeline(this.liveTimeline, { backwards: true, limit: 1 });
+                if (this.replyCount === 0 && this.rootEvent) {
+                    this.timelineSet.addEventsToTimeline([this.rootEvent], true, this.liveTimeline, null);
+                    this.liveTimeline.setPaginationToken(null, Direction.Backward);
+                } else {
+                    await this.client.paginateEventTimeline(this.liveTimeline, {
+                        backwards: true,
+                        limit: Math.max(1, this.length),
+                    });
+                }
```
Comment (MadLittleMods) [inline src/models/thread.ts:369]: What was wrong before that caused https://github.com/vector-im/element-web/issues/23934?

What changed so https://github.com/vector-im/element-web/issues/23934 now works?

If there is some tricky order of operations here, we should explain why/what, etc

Comment (MadLittleMods) [reply src/models/thread.ts:369]: @justjanne ^

Comment (justjanne) [reply src/models/thread.ts:369]: @MadLittleMods I'm currently on vacation, I'll address it in January.

**Pull request merged**